### PR TITLE
Add support for the Jenkins Heavy Job plugin

### PR DESCRIPTION
**Pull request by cottsay**
This plugin will need to be installed at the same time this PR is merged to avoid unnecessary diffs in the job configurations.

The intended use case is to allow some jobs (namely the nightly CI jobs) to take up more executor slots on the workers to avoid resource contention with other jobs.

### CODE REVIEW

**Comment by dirk-thomas** [comment]
Does the XML snippet need to be present in every job even if the job doesn't set a custom value?

**Comment by cottsay** [comment]
> Does the XML snippet need to be present in every job even if the job doesn't set a custom value?

Unfortunately, yes. Once the plugin is installed, all jobs get the snippet.

**Comment by dirk-thomas** [comment]
Can you please create a corresponding PR then to add the plugin to the `buildfarm_deployment` repo.

**Comment by cottsay** [comment]
> Can you please create a corresponding PR then to add the plugin to the buildfarm_deployment repo.

Done: ros-infrastructure/buildfarm_deployment#242

File: doc/configuration_options.rst [hunk 1]
```diff
@@ -458,6 +458,11 @@ The following options are valid in version ``1`` (beside the generic options):
 * ``jenkins_job_upstream_triggers``: names of other CI jobs which, when
   built with a stable or unstable result, should trigger this job to be built.
 
+* ``jenkins_job_weight``: the number of executors on a worker which are
+  required to execute the job.
+  Default is ``1``.
+  Activates the Jenkins Heavy Job plugin.
```
Comment (dirk-thomas) [inline doc/configuration_options.rst:464]: Since the plugin needs to be present anyway maybe rephrase this from "Activates ..." to just "Uses ..." or "Relies on ..."?

Comment (cottsay) [reply doc/configuration_options.rst]: 2f0a74a

**Comment by ros-discourse** [comment]
This pull request has been mentioned on **ROS Discourse**. There might be relevant details there:

https://discourse.ros.org/t/jenkins-heavy-job-plugin/16265/1
